### PR TITLE
add API for dynamic-value foods

### PR DESCRIPTION
**Pull request by LemmaEOF**
- add and implement a DynamicFood interface for foods that have restoration values based on their item stack

### CODE REVIEW

**Comment by squeek502** [comment]
Any reason it shouldn't be something like:

```java
int getDynamicHunger(PlayerEntity player, ItemStack stack);
float getDynamicSaturation(PlayerEntity player, ItemStack stack);
```

?

Seems like that would be more flexible/easier to understand, and it'd be more similar to the AppleCore/The Spice of Life way of doing things.

**Comment by LemmaEOF** [comment]
Yeah, that sounds better actually. Lemme change that quick.

File: java/squeek/appleskin/helpers/FoodHelper.java [hunk 1]
```diff
@@ -52,7 +52,11 @@ public static BasicFoodValues getDefaultFoodValues(ItemStack itemStack)
 		FoodItemSetting itemFood = itemStack.getItem().getFoodSetting();
 		int hunger = itemFood.getHunger();
 		float saturationModifier = itemFood.getSaturationModifier();
-
+		if (itemStack.getItem() instanceof DynamicFood) {
```
Comment (squeek502) [inline java/squeek/appleskin/helpers/FoodHelper.java]: This should go in `getModifiedFoodValues` instead. Items can already modify their default values using the standard methods, right?

File: java/squeek/appleskin/helpers/DynamicFood.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * An interface for foods whose hunger/saturation restoration can vary by the ItemStack.
+ */
+public interface DynamicFood
+{
+	/**
+	 * @param stack the stack to check
+	 * @return how much hunger an instance of this food will restore in total, including base restoration.
+	 */
+	int getDynamicHunger(ItemStack stack);
```
Comment (squeek502) [inline java/squeek/appleskin/helpers/DynamicFood.java]: These should take a PlayerEntity parameter too, for things like The Spice of Life that modify food values on a per-player basis.

**Review by squeek502 — CHANGES_REQUESTED** [review]

File: java/squeek/appleskin/helpers/DynamicFood.java (hunk 2 above)
Comment (LemmaEOF) [reply java/squeek/appleskin/helpers/DynamicFood.java]: Ah yeah, I guess the calls of this code should be moved to the PlayerEntity-accepting section of the FoodHelper.

**Review by squeek502 — APPROVED** [review]

**Comment by squeek502** [comment]
Looks good, thanks!

**Pull request merged**